nmoscommon/auth/nmos_auth.py

```diff
@@ -42,83 +43,109 @@
-        if contentType == "application/json":
-            cert = cert_resp.json()
+        if "application/json" in jwk_resp.headers['content-type']:
             try:
-                if len(cert) > 1:
-                    self.logger.writeWarning("Multiple certificates at Endpoint. Returning First Instance.")
-                cert = cert[0]
-                return cert
+                jwks = jwk_resp.json()
+                if "keys" in jwks:
+                    jwks_keys = jwks["keys"]
+                    self.logger.writeInfo("JSON Web Key Set located at /jwks endpoint.")
+                else:
+                    jwks_keys = jwks
+                return jwks_keys
             except Exception as e:
-                self.logger.writeError("Error: {}. Endpoint contains: {}".format(str(e), cert))
+                self.logger.writeError("Error: {}. JWK endpoint contains: {}".format(str(e), jwk_resp.text))
                 raise
         else:
-            self.logger.writeError("Incorrect Content-Type. Expected 'application/json but got {}".format(contentType))
+            self.logger.writeError("Incorrect Content-Type. Expected 'application/json but got {}".format(
+                jwk_resp.headers['content-type']))
             raise ValueError
 
     def getCertFromFile(self, filename):
         try:
             if filename is not None:
                 with open(filename, 'r') as myfile:
-                    cert_data = myfile.read()
-                    self.certificate = cert_data
-                    return cert_data
+                    cert = myfile.read()
+                    return cert
         except OSError:
-            self.logger.writeError("File does not exist or you do not have permission to open it")
+            self.logger.writeError("File {} does not exist or you do not have permission to open it".format(
+                filename))
             raise
 
-    def extractPublicKey(self, certificate):
-        crtObj = crypto.load_certificate(crypto.FILETYPE_PEM, certificate)
-        pubKeyObject = crtObj.get_pubkey()
-        pubKeyString = crypto.dump_publickey(crypto.FILETYPE_PEM, pubKeyObject)
-        if pubKeyString is None:
-            self.logger.writeError("Public Key could not be extracted from certificate")
-            raise ValueError
-        else:
-            return pubKeyString.decode('utf-8')
+    def getPublicKeyString(self, key_class):
+        serialised_key = key_class.public_bytes(
+            encoding=serialization.Encoding.PEM,
+            format=serialization.PublicFormat.SubjectPublicKeyInfo
+        )
+        return serialised_key.decode('utf-8')
+
+    def extractPublicKey(self, key_containing_object):
+        """Extracts a key from the given parameter. A list or a dict object will be treated as a JWK structure.
+        A string will be treated like a X509 certificate"""
+        try:
+            if isinstance(key_containing_object, dict):
```

Comment:
Hopefully this can be simplified as a result of using the defined /jwks structure.